chia/server/node_discovery.py

```diff
@@ -351,7 +351,7 @@ async def _connect_to_peers(self, random) -> None:
                     # only consider very recently tried nodes after 30 failed attempts
                     # attempt a node once per 30 minutes if we lack connections to increase the chance
                     # to try all the peer table.
-                    if now - info.last_try < 1800 and tries < 30:
+                    if now - info.last_try < 1800:
```

Comment:
my understanding is that this will make us more eager to ask the introducer for more peers. If all `max_tries` peers we get from the address manager are not eligible for a connection attempt, we take that to indicate that the address manager doesn't have a lot of peers in it, and that we may have already tried most of them. That's all fine (a comment to that effect would be nice. I find it a bid odd that we deduce this information about the peers we have, rather than just knowing. The `address_manager` knows).

So, in this case we contact the introducer (and this coroutine waits for a response (which will add peers to the `address_manager`).

We enter this loop again. We still don't have any peer connections, so `len(groups)` is 0 and `sleep_interval` is 0.1.

We try at most 10 more peers. If we still can't find a connection candidate (perhaps `address_manager` had so many recently-tried peers in it that it diluted the new ones we got). We'll keep looping through here, contacting the introducer and trying 10 more peers. Still only sleeping 0.1 second in between tries.

Or maybe we actually initiate a connection. We won't contact the introducer again, but we also won't actually have any connections still (because establishing a connection is not instantaneous). Let's say it takes 5 seconds to connect and SSL-handshake. That means 50 more turns through this loop.

that might not be that big deal.

Reply:
I guess the greater risk is that, while waiting for a connection attempt to complete, if we don't find any more connect candidates, every 10 attempts will make us hit the introducer again. This may also not be a big issue, since most likely the user's ISP has cached the DNS response, and it won't cause load on our DNS introducer. But it's not very elegant.